daemon/network.go

```diff
@@ -117,15 +117,16 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	return c.NewNetwork(driver, name, nwOptions...)
 }
 
-func getIpamConfig(data []network.IPAMConfig) ([]*libnetwork.IpamConf, []*libnetwork.IpamConf, error) {
+func getIpamConfig(data network.IPAM) ([]*libnetwork.IpamConf, []*libnetwork.IpamConf, error) {
 	ipamV4Cfg := []*libnetwork.IpamConf{}
 	ipamV6Cfg := []*libnetwork.IpamConf{}
-	for _, d := range data {
+	for _, d := range data.Config {
 		iCfg := libnetwork.IpamConf{}
 		iCfg.PreferredPool = d.Subnet
 		iCfg.SubPool = d.IPRange
 		iCfg.Gateway = d.Gateway
 		iCfg.AuxAddresses = d.AuxAddress
+		iCfg.Options = data.Options
```

Comment:
No problem. I forgot I left those in I will have those changed. Also yes I was going to suggest making a change to the NetworkOptionIpam to accept the changes correctly. 

I have no problem with you pushing changes in libnetwork and getting them vendored in. If you can just mention this pull just so I can follow the changes that would be great.